docs/spec/slashing/state-machine.md

```diff
@@ -0,0 +1,174 @@
+# State Machine Interaction Overview
+
+## Conceptual overview
+
+### States
+
+At any given time, there are any number of validator candidates registered in the state machine.
```

Comment:
`validator candidates` -> `bonded validators`
also generally remove use of candidates throughout

Reply:
the term `validators` includes validators in all states - so if you want to say `validators/bonded validators` you might as well just say `validators`

Reply:
Exactly; I mean "validators" instead of "validator candidates" - which I did I think (lmk if I missed any).